Domains/1-Common/DistributionSystems/Released/DistributionSystems.01.00.02.ecschema.xml

```diff
@@ -0,0 +1,152 @@
+        <BaseClass>bis:ElementRefersToElements</BaseClass>
+        <Source multiplicity="(0..*)" polymorphic="true" roleLabel="uses">
+            <Class class="IDistributionElementType"/>
+        </Source>
+        <Target multiplicity="(0..*)" polymorphic="true" roleLabel="is used by">
+            <Class class="DistributionPortType"/>
+        </Target>
+        <ECProperty propertyName="PortCount" typeName="int" displayLabel="Port Count" description="Port-instance count of the referenced DistributionPortType used by the IDistributionElementType."/>
+    </ECRelationshipClass>
+
+    <ECEnumeration typeName="PortDirection" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Undefined" value="0" displayLabel="Undefined"/>
+        <ECEnumerator name="Input" value="1" displayLabel="Input"/>
+        <ECEnumerator name="Output" value="2" displayLabel="Output"/>
+        <ECEnumerator name="InputOrOutput" value="3" displayLabel="Input Or Output"/>
+        <ECEnumerator name="InputAndOutput" value="4" displayLabel="Input And Output"/>
+    </ECEnumeration>
+
+    <ECEntityClass typeName="DistributionPort" modifier="Abstract" displayLabel="Distribution Port" description="An inlet or outlet of a distribution element through which a particular substance may flow.">
+        <BaseClass>bis:SpatialLocationElement</BaseClass>
+        <ECProperty propertyName="Description" typeName="string" displayLabel="Description" description="A human-readable description of the Distribution Port"/>
+        <ECProperty propertyName="Direction" typeName="PortDirection" displayLabel="Direction" description="Direction of the port: e.g. Input, Output, InputOrOutput, InputAndOutput."/>
+    </ECEntityClass>
+    
+    <ECRelationshipClass typeName="DistributionElementOwnsDistributionPorts" strength="embedding" modifier="Abstract" description="A relationship that relates child distribution ports to parent distribution element.">
+        <BaseClass>bis:ElementOwnsChildElements</BaseClass>
+        <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="true">
+            <Class class="IDistributionElement"/>
+        </Source>
+        <Target multiplicity="(1..*)" roleLabel="is owned by" polymorphic="true">
+            <Class class="DistributionPort"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <ECEntityClass typeName="DistributionPortType" modifier="Abstract" displayLabel="Distribution Port Type" description="Defines a shared set of properties whose values vary per-type of IDistributionElement rather than per-instance.">
+        <BaseClass>bis:SpatialLocationType</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="PortConnection" modifier="None" displayLabel="Port Connection" description="An element that indicates a connection between 2 distribution ports.">
```

Comment:
The PortConnection itself doesn't usually have discipline-specific semantics. Its need is primarily to "objectify" a connection between ports so one can target it with relationships (e.g. "PhysicalElementRealizesConnection"). That's why it made sense to us to offer such class as concrete to be use as-is by disciplines, rather than abstract in order to force every discipline to subclass.